### DISCUSSION TITLE
Client reports a different topic than wallet

### DISCUSSION THREAD

**Discussion by stefandunca** [WalletConnect/walletconnect-monorepo#3994] [Developer Support] [answered]
Hello Wallet Connect team,

I experienced an issue with non-matching pairing topics in a pairing client-wallet session. Please advise us on what might go wrong in there.

Facts:
- I'm testing our integration with `web3wallet` SDK using [react test app](https://react-app.walletconnect.com/) as client
- the wallet SDK side runs `"@walletconnect/web3wallet": "^1.9.0"` (downgraded from 1.9.2 as per this discussion [#3291](https://github.com/orgs/WalletConnect/discussions/3291) but it didn't help)
- the client calls (`core.pairing.getPairings()`) report the same pairing topic as presented in the pairing URI, but the client test app reports a different `topic` in the event `session_delete`
- also, a `disconnect` from the wallet side doesn't have any effect on the client side
- I experience the same with the `https://se-sdk-dapp.vercel.app/` test app
- the client requests valid data (addresses and chains) in `session_request`

Following are the logs from establishing a pairing from the wallet and disconnecting from the client side
the wallet uses the topic from the pairing URI `c5726f7e3529bc386318878af7c9de7dc85c0222d3e95f2c6a4a82c32b2bf9b1 `
client reports a different topic `43323911273a1167f8ca98ceaa55627eb5715e08cdf77bd0e257de5d2af51df9`

```json
// Pairing URI
"wc:c5726f7e3529bc386318878af7c9de7dc85c0222d3e95f2c6a4a82c32b2bf9b1@2?relay-protocol=irn&symKey=d3ef6d595efaf357fca3a19f22be3009a7ecca23c75315eb30ff47b4ecab1e92"
// SDK `session_proposal` event
{
  "id": 1701250738757624,
  "params": {
    "expiry": 1701251101,
    "id": 1701250738757624,
    "optionalNamespaces": {
      "eip155": {
        "chains": [
          "eip155:5"
        ],
        "events": [],
        "methods": [
          "eth_signTransaction",
          "eth_sign",
          "eth_signTypedData",
          "eth_signTypedData_v4"
        ]
      }
    },
    "pairingTopic": "c5726f7e3529bc386318878af7c9de7dc85c0222d3e95f2c6a4a82c32b2bf9b1",
    "proposer": {
      "metadata": {
        "description": "React App for WalletConnect",
        "icons": [
          "https://avatars.githubusercontent.com/u/37784886"
        ],
        "name": "React App",
        "url": "https://react-app.walletconnect.com",
        "verifyUrl": "https://verify.walletconnect.com"
      },
      "publicKey": "c7a7ba49521bca961e86181d00182ba5ed8677763c5547e5c46250410c0fb060"
    },
    "relays": [
      {
        "protocol": "irn"
      }
    ],
    "requiredNamespaces": {
      "eip155": {
        "chains": [
          "eip155:5"
        ],
        "events": [
          "chainChanged",
          "accountsChanged"
        ],
        "methods": [
          "eth_sendTransaction",
          "personal_sign"
        ]
      }
    }
  },
  "verifyContext": {
    "verified": {
      "origin": "https://react-app.walletconnect.com",
      "validation": "UNKNOWN",
      "verifyUrl": "https://verify.walletconnect.com"
    }
  }
}
// SDK.approveSession:
// id param
1701250738757624
// supportedNamespaces param
{
  "eip155": {
    "chains": [
      "eip155:5"
    ],
    "methods": [
      "personal_sign",
      "eth_sendTransaction",
      "eth_signTransaction",
      "eth_sign",
      "eth_signTypedData",
      "eth_signTypedData_v4"
    ],
    "events": [
      "chainChanged",
      "accountsChanged"
    ],
    "accounts": [
      "eip155:5:0xBd54A96c0Ae19a220C8E1234f54c940DFAB34639"
    ]
  }
}
// SDK.approveSession response
{
  "error": "",
  "result": {}
}

// SDK.getPairings
[
    {
      "active": true,
      "expiry": 1703842806,
      "peerMetadata": {
        "description": "React App for WalletConnect",
        "icons": [
          "https://avatars.githubusercontent.com/u/37784886"
        ],
        "name": "React App",
        "url": "https://react-app.walletconnect.com",
        "verifyUrl": "https://verify.walletconnect.com"
      },
      "relay": {
        "protocol": "irn"
      },
      "topic": "c5726f7e3529bc386318878af7c9de7dc85c0222d3e95f2c6a4a82c32b2bf9b1"
    }
  ]
// SDK "session_delete"
{
  "id": 1701250825547896,
  "topic": "43323911273a1167f8ca98ceaa55627eb5715e08cdf77bd0e257de5d2af51df9"
}
```

Thank you

**Comment by glitch-txs** [comment]
Hey it could be caused by different reasons
- first make sure you're not instantiating Web3Wallet more than once, same thing for the events. 
- Also we added some error descriptions in our docs, let me know if it helps. https://docs.walletconnect.com/web3wallet/wallet-usage?platform=react-native
- Another source for testing I'd recommend is https://lab.web3modal.com/
- And I'd highly recommend using the latest version as well

**Reply by stefandunca** [reply]
Thanks for your answer. I will take them into consideration.

**Reply by stefandunca** [reply, marked as answer]
The answer was in this post: https://medium.com/walletconnect/walletconnect-v2-0-protocol-whats-new-3243fa80d312

We used to track pairing by the pairing topic. However, the solution is to track active sessions and use the session topic.
